### DISCUSSION TITLE
Gizmos seem slow to draw

### DISCUSSION THREAD

**Discussion by franklinblanco** [bevyengine/bevy#12005] [Q&A]
Hi, I'm trying to draw gizmos for a laser, and it seems to be slow at drawing the actual line. Can't figure it out, already tried to run it on a FixedUpdate of 240hz and 100hz and 60hz, nothing has solved it.

Preview:

https://github.com/bevyengine/bevy/assets/72827992/28df24cf-5fe1-495a-9f27-a22e75ee0bc4

**Comment by franklinblanco** [comment]
Additional Context: 

### Registering the system:
```rust
app.add_systems(FixedUpdate, draw_debug_bore_laser)
        .insert_resource(Time::<Fixed>::from_hz(240.0))
```

### System:
```rust
pub fn draw_debug_bore_laser(
    query: Query<&GlobalTransform, (With<InPlayerHandsMarker>, With<FiringPoint>)>,
    mut gizmos: Gizmos,
) {
    for transform in query.iter() {
        let max_toi = transform.forward() * 20.0;
        gizmos.line(
            transform.translation(),
            transform.translation() + max_toi,
            Color::Rgba { red: 2.5, green: 0.0, blue: 0.0, alpha: 1.0 },
        );
    }
}
```

**Comment by Ihazevich** [comment]
I had a similar issue and I solved it by making sure that the gizmos are drawn after the camera is moved.

**Reply by franklinblanco** [reply]
Hi, thanks for the reply. In my case the camera doesn't move, it does rotate though.

**Comment by ikbencasdoei** [comment]
Its most likely because you added the draw_debug_bore_laser system to the FixedUpdate schedule which doesn't always run before the screen is rendered. I would suggest adding it to the normal Update schedule.

**Reply by franklinblanco** [reply]
Thanks for writing, I initially tried the Update schedule and that's why I tried switching to the FixedUpdate. Now it's on the Update schedule and it still runs like that.

**Reply by ikbencasdoei** [reply]
Okay because in my experience the gizmos draw instantly, so there are only a few other things I can think of:
- The draw_debug_bore_laser runs before the system that updates the position/rotation, which causes a one frame delay.
You could fix this using:
```rust
app.add_systems(Update, draw_debug_bore_laser.after(/*insert FiringPoint rotation system or label*/))
```
Or,
- There is somewhere a delay updating the globaltransform.

**Reply by suptimal** [reply]
@franklinblanco after moving the system to PostUpdate this issue was fixed for me.

 ```rust
app.add_systems(PostUpdate, draw_debug_bore_laser)
 ```
